src/util/painterscope.h

```diff
@@ -0,0 +1,28 @@
+//
+// Created by Ferran Pujol Camins on 2019-06-22.
+//
+
+#pragma once
+
+#include <QPainter>
+
+// This class provides RAII style management of a QPainter properties.
+//
+// PainterScope will save the painter state on creation, and restore it
+// on destruction.
+class PainterScope {
```

Comment:
QMutexLocker wraps a pointer. Even though a reference is more appropriate, I would suggest to pass and wrap a pointer here just for consistency.

PainterScope should be final (class) and non-copyable (ctor).

Reply:
Suggestions addressed. Thanks for the review.